<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <title></title>
</head>
<body>
    <script>
        console.log(1);
        console.log(2);
        console.log(3);

        console.log(4);
        //debugger;
        let resolver = function (resolve, reject) {
            setTimeout(function () {
                console.log(5);
                resolve();
                //reject("Error");
            }, 1000);
        }
        let print5Promise = new Promise(resolver);
        print5Promise.then(function () {
            console.log(6);
        });
        print5Promise.catch((error) => {
            console.log(error + ", can't find number");
        });
        console.log(print5Promise);
        //console.log(6);

        async function print56() {
            await print5Promise;
            console.log(7);
        }
        async function main() {
            try {
                await print56();
            } catch (error) {
                console.log(error);
            }
        }
        main();

    </script>
</body>
</html>